src/test/operators/jit_operator/operators/jit_limit_test.cpp

```diff
@@ -0,0 +1,62 @@
+#include "../../../base_test.hpp"
+#include "operators/jit_operator/operators/jit_limit.hpp"
+
+namespace opossum {
+
+namespace {
+
+// Mock JitOperator that records whether tuples are passed to it
+class MockSink : public AbstractJittable {
+ public:
+  MockSink() : AbstractJittable(JitOperatorType::Write) {}
+
+  std::string description() const final { return "MockSink"; }
+
+ private:
+  void _consume(JitRuntimeContext& context) const final {}
+};
+
+// Mock JitOperator that passes on individual tuples
+class MockSource : public AbstractJittable {
+ public:
+  MockSource() : AbstractJittable(JitOperatorType::Read) {}
+
+  std::string description() const final { return "MockSource"; }
+
+  void emit(JitRuntimeContext& context) { _emit(context); }
+
+ private:
+  void _consume(JitRuntimeContext& context) const final {}
+};
+
+}  // namespace
+
+class JitLimitTest : public BaseTest {};
+
+TEST_F(JitLimitTest, FiltersTuplesAccordingToLimit) {
+  const uint32_t chunk_size{123};
+
+  JitRuntimeContext context;
+  context.chunk_size = chunk_size;
+  context.limit_rows = 2;
+
+  auto source = std::make_shared<MockSource>();
+  auto limit = std::make_shared<JitLimit>();
+  auto sink = std::make_shared<MockSink>();
```

Comment:
Would it make sense for the sink to count the number of tuples it received?

Reply:
That would be possible. There are different possibilities on how to implement a Limit operator. I chose the option to decrement the counter within the Limit operator as this only requires one access to the context (`--context.limit_rows`) for every tuple.

Reply:
I mean just for the test.